src/MentionsInput.js

```diff
@@ -541,15 +541,42 @@ const MentionsInput = React.createClass({
     // neglect async results from previous queries
     if(queryId !== this._queryId) return;
 
+    var type = mentionDescriptor.props.type;
     var update = {};
-    update[mentionDescriptor.props.type] = {
-      query: query,
-      mentionDescriptor: mentionDescriptor,
-      querySequenceStart: querySequenceStart,
-      querySequenceEnd: querySequenceEnd,
-      results: suggestions,
-      plainTextValue: plainTextValue
-    };
+
+    // If mention doesn't specify type try suggestion type
+    if(!type) {
+      var groups = {}
```

Comment:
In my scenario, there is only one mention source but the mentionable data comes from around 12 different sources. This is done to give the user an ability to mention any type of entity with the same trigger while allowing us to define behavior for different mentioned types.